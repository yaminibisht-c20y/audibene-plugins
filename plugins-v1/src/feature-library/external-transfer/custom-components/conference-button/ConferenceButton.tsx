import * as React from 'react';
import {
  Actions,
  IconButton,
  TaskHelper,
  withTheme
} from '@twilio/flex-ui';

class ConferenceButton extends React.PureComponent<any, any> {
    handleClick = () => {
        Actions.invokeAction('SetComponentState', {
        name: 'ConferenceDialog',
        state: { isOpen: true }
        });
    }

    render() {
        const isLiveCall = TaskHelper.isLiveCall(this.props.task);

        return (
        <React.Fragment>
            <IconButton
            icon="Add"
            disabled={!isLiveCall}
            onClick={this.handleClick}
            themeOverride={this.props.theme.CallCanvas.Button}
            title={"Add Conference Participant"}
            />
        </React.Fragment>
        );
    }
}

export default withTheme(ConferenceButton);