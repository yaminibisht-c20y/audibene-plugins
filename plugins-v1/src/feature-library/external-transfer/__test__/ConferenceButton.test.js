import React from 'react';
import { shallow } from 'enzyme';
import ConferenceButton from "../custom-components/conference-button";

jest.mock("@twilio/flex-ui",()=>({
    TaskHelper: {
        isLiveCall: jest.fn(()=>{return Promise.resolve(true)})
    },
    withTheme: (M) =>{ return M},
    Actions: {
        invokeAction: jest.fn()
    }
}))

describe("test ConferenceButton component", () => {
    let props = {
        task:{

        },
        theme:{
            CallCanvas:{
                Button: false
            }
        }
    }
    it("component should render successfully", () => {
        const wrapper = shallow(<ConferenceButton {...props}/>);
        expect(wrapper).toMatchSnapshot();
    })

    it("test function",()=>{
        const wrapper = shallow(<ConferenceButton {...props}/>);
        const instancew = wrapper.instance();
        instancew.handleClick();
    })
})