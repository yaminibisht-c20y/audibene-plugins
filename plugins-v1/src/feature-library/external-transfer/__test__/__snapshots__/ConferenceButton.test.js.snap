// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test ConferenceButton component component should render successfully 1`] = `ShallowWrapper {}`;
